src/item_factory.cpp

```diff
@@ -2097,6 +2112,15 @@ void Item_factory::load_basic_info( JsonObject &jo, itype &def, const std::strin
         }
     }
 
+    if( jo.has_array( "batteries" ) ) {
+        def.batteries.clear();
+
+        JsonArray bats = jo.get_array( "batteries" );
+        for( size_t i = 0; i < bats.size(); ++i ) {
+            def.batteries.insert( bats.get_string( i ) );
```

Comment:
Would be nice if you added consistency checks for those ids in `Item_factory::check_definitions`.

Reply:
Added checks for battery items. Items loading batteries is no longer in this branch.